Task-54083 : Add documentation for antibrute force system (#314)
exoplatform/exo-documentation: docs/Security.rst

       Information about the file location and steps to update the
       "Remember My Login" password encryption key.

    -  :ref:`Anti Brute Force <Security.AntibruteForce>`

>        To configure antibrute force system.

Reviewer: ```suggestion
       To configure the mechanim that protects against brute force attacks on password authentication.
```

.. note:: Updating the password encryption key causes the invalidation of existing tokens, so the users must re-login.

.. _Security.AntibruteForce:

=================================
Login Brute Force Attacks Protection
=================================

> To prevent an attack based on brute force on login/password form, a built-in protection mechanism exists.

Reviewer: ```suggestion
To prevent an attack based on brute force on login/password form, a built-in protection mechanism exists : 
```

> When multiple failed login attempts occur in a short time, the target user account is temporarily locked for a few minutes.

Reviewer: ```suggestion
when multiple failed login attempts occur in a short time, the target user account is temporarily locked for a few minutes.
```

> When an account is locked, the user can immediately unlock it by resetting its password through a __forgot password__ request.

Reviewer: ```suggestion
When an account is locked, the user can immediately unlock it by resetting its password through a **forgot password** request.
```

.. note:: Updating the password encryption key causes the invalidation of existing tokens, so the users must re-login.

.. _Security.AntibruteForce:

=================================

> Anti Brute Force Login Protection

Reviewer: ```suggestion
Login Brute Force Attacks Protection
```

=================================

> To prevent an attack based on bruteforce on login form, eXo Platform have an antibrute force system.

Reviewer: ```suggestion
To prevent an attack based on brute force on login/password form, a built-in protection mechanism exists.
```

> After few failed logins attempts in a short time, the concerned user account is locked for some minutes.

Reviewer: ```suggestion
When multiple failed login attempts occur in a short time, the target user account is temporarily locked for a few minutes.
```

> When an account is locked, the user can make a forgot password request. After resetting his password, the account is unlocked

Reviewer: ```suggestion
When an account is locked, the user can immediately unlock it by resetting its password through a __forgot password__ request.
```

> and the user can login normally.

Reviewer: ```suggestion
```

> There are two properties to configure this feature. To modify it, you can add it in ``exo.properties``.

Reviewer: ```suggestion
Two properties control the brute force attack protection mechanism. To configure it, you can add them in ``exo.properties``.
```

> The following property allow to configure the number of attempts before the account is locked. The default value is 5 attempts

Reviewer: ```suggestion
The following property determines the number of unsuccessful login attempts before the account is locked. The default value is 5 attempts
```

   ::

		exo.authentication.antibruteforce.maxAuthenticationAttempts=5

> The following property allow to configure the number of minutes the account is locked. The default value is 10 minutes.

Reviewer: ```suggestion
The following property determines how long (in minutes) an account is locked when the protection mechanism is triggered. The default value is 10 minutes.
```
